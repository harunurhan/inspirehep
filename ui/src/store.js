import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
/* eslint-disable import/no-extraneous-dependencies */
import { composeWithDevTools } from 'redux-devtools-extension';

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import keepPreviousUrlMiddleware from './middlewares/keepPreviousUrl';
import {
  reHydrateRootStateFromStorage,
  createPersistToStorageMiddleware,
} from './middlewares/statePersister';
import searchDispatcherMiddleware from './middlewares/searchDispatcher';
import redirectToErrorPageMiddleware from './middlewares/redirectToErrorPage';
import actionTrackerMiddleware from './middlewares/actionTracker';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const reducersToPersist = ['ui', 'user'];

const PROD_MIDDLEWARES = [
  reduxRouterMiddleware,
  queryParamsParserMiddleware,
  keepPreviousUrlMiddleware,
  createPersistToStorageMiddleware(reducersToPersist),
  searchDispatcherMiddleware,
  redirectToErrorPageMiddleware,
  actionTrackerMiddleware,
  thunkMiddleware,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(...PROD_MIDDLEWARES);
  }
  return applyMiddleware(...DEV_MIDDLEWARES);
};

export const store = createStore(
  reducers,
  reHydrateRootStateFromStorage(reducersToPersist),
  composeWithDevTools(withMiddlewares())
);
