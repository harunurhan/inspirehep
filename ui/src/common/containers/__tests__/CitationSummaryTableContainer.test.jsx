import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { getStoreWithState } from '../../../fixtures/store';
import CitationSummaryTableContainer from '../CitationSummaryTableContainer';

describe('CitationSummaryTableContainer', () => {
  it('pass props from state', () => {
    const store = getStoreWithState({
      citations: fromJS({
        loadingCitationSummary: false,
        errorCitationSummary: null,
        citationSummary: {
          citations: {
            buckets: {
              all: { name: 'citeable' },
              published: { name: 'published' },
            },
          },
          'h-index': {
            value: { name: 'h-index' },
          },
        },
      }),
    });
    const wrapper = shallow(<CitationSummaryTableContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
