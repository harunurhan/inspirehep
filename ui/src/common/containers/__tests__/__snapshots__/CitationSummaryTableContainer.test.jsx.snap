// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationSummaryTableContainer pass props from state 1`] = `
<CitationSummaryTable
  citeableBucket={
    Immutable.Map {
      "name": "citeable",
    }
  }
  dispatch={[Function]}
  error={null}
  hIndex={
    Immutable.Map {
      "name": "h-index",
    }
  }
  loading={false}
  publishedBucket={
    Immutable.Map {
      "name": "published",
    }
  }
  renderNumberOfCiteablePapers={[Function]}
  renderNumberOfPublishedPapers={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
